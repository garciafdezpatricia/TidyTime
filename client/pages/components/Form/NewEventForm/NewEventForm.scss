@use "../../../../styles/variables" as var;

.create-event-form {
    display: flex;
    flex-direction: column;
}