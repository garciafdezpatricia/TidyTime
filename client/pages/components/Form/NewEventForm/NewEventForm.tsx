export default function NewEventForm() {




    return (
        <form className="create-event-form">
            <label>Title</label>
            <input></input>
            <label>Description</label>
            <input>
            </input>
            <label>Time</label>
            <label>Color</label>
        </form>
    )
}