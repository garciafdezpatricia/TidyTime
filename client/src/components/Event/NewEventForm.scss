@use "../../../styles/variables" as *;


.new-event-form {
    padding: 1.5rem 1.5rem 1.5rem 1rem;
    display: flex;
    flex-direction: column;
    gap: .5rem;
    width: 100%;
    box-sizing: border-box;
    font-size: .8rem;

    @media screen and (min-width: 320px) and (max-width: 768px) {
        gap: 1.2rem;
    }

    label {
        display: flex;
        justify-content: space-between;

        @media screen and (min-width: 320px) and (max-width: 768px) {
            gap: .5rem;
            align-items: center;
        }
    }

    input[type="text"], textarea {
        border-radius: .3rem;
        padding: .2rem .4rem;
        border: .1rem solid $light-grey;
        font-size: .8rem;
    }

    textarea {
        resize: none;
    }

    input[type="datetime-local"] {
        border-radius: .3rem;
        padding: 0 .4rem;
        border: .1rem solid $light-grey;
        width: fit-content;
        font-size: .8rem;
    }

    input[type="color"] {
        width: 2rem;
        background-color: $hover-green;
        padding: .2rem;
        border: none;
    }

    .create-new-event-btn {
        width: fit-content;
        align-self: center;
        border: .1rem solid $medium-grey;
        color: $black;
        background-color: $primary-grey;
        padding: 0 .4rem;
        display: flex;
        gap: .5rem;
        &:hover {
            background-color: $hover-green;
            border-color: $primary-green;
        }
    }
}