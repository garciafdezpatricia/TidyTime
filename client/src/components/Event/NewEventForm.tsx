import { useTranslation } from "react-i18next";
import ComboBox from "../ComboBox/ComboBox";
import { MutableRefObject, useState } from "react";

export interface Props {
    startDate?: Date, //yyyy-mm-ddThh:mm
    endDate?: Date, //yyyy-mm-ddThh:mm
    titleRef: MutableRefObject<any>,
    infoRef: MutableRefObject<any>,
    fromDateRef: MutableRefObject<any>,
    toDateRef: MutableRefObject<any>,
    onColorChange: (arg?:any) => void | any; 
}

export default function NewEventForm({startDate, endDate, titleRef, infoRef, fromDateRef, toDateRef, onColorChange} : Props) {
    const { t } = useTranslation();
    const start = startDate 
        ? new Date(startDate.getTime() - (startDate.getTimezoneOffset() * 60000)).toISOString().slice(0, 16) 
        : "";
    const end = endDate
    ? new Date(endDate.getTime() - (endDate.getTimezoneOffset() * 60000)).toISOString().slice(0, 16) 
    : "";  
    const [selectedColor, setSelectedColor] = useState("#3E5B41");

    const handleOptionChange = (color:string) => {
        setSelectedColor(color);
        onColorChange(color);
    }

    return (
        <article className="new-event-form">
            <input ref={titleRef} placeholder={t('calendar.calendarMenu.addEvent.addTitle')} type="text" />
            <textarea ref={infoRef} placeholder={t('calendar.calendarMenu.addEvent.addNotes')} />
            <label>{t('calendar.calendarMenu.addEvent.from')}<input ref={fromDateRef} type="datetime-local" defaultValue={start} /> </label>
            <label>{t('calendar.calendarMenu.addEvent.to')}<input ref={toDateRef} type="datetime-local" defaultValue={end} /> </label>
            <ComboBox 
                text={t('calendar.calendarMenu.addEvent.color')}
                colors={["#e13535", "#d37373", "#ffa500", "#3E5B41", "#3bb4ff"]}
                checkedOption={""}   
                onOptionChange={handleOptionChange} 
            />
        </article>
    )
}