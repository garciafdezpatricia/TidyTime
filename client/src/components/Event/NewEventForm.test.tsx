import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import NewEventForm from '@/src/components/Event/NewEventForm';

describe('NewEventForm', () => {
    it('renders without crashing', () => {
    const { container } = render(
        <NewEventForm 
        titleRef={{current: null}} 
        infoRef={{current: null}} 
        fromDateRef={{current: null}} 
        toDateRef={{current: null}} 
        onColorChange={() => {}} 
        />
    );
    expect(container).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(
            <NewEventForm 
            titleRef={{current: null}} 
            infoRef={{current: null}} 
            fromDateRef={{current: null}} 
            toDateRef={{current: null}} 
            onColorChange={() => {}} 
            />
        );
        expect(asFragment()).toMatchSnapshot();
      });
  });