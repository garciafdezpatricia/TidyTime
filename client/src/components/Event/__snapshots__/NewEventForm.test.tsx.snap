// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewEventForm matches snapshot 1`] = `
<DocumentFragment>
  <article
    class="new-event-form"
  >
    <input
      placeholder="calendar.calendarMenu.addEvent.addTitle"
      type="text"
    />
    <textarea
      placeholder="calendar.calendarMenu.addEvent.addNotes"
    />
    <label>
      calendar.calendarMenu.addEvent.from
      <input
        data-testid="from-date"
        type="datetime-local"
        value=""
      />
    </label>
    <label>
      calendar.calendarMenu.addEvent.to
      <input
        data-testid="to-date"
        type="datetime-local"
        value=""
      />
    </label>
    <div
      class="combobox"
    >
      <div
        class="combobox-header"
      >
        calendar.calendarMenu.addEvent.color
        <svg
          color="#000000"
          cursor="pointer"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(0, 0, 0);"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M48 256c0 114.9 93.1 208 208 208s208-93.1 208-208S370.9 48 256 48 48 141.1 48 256zm289.1-43.4c7.5-7.5 19.8-7.5 27.3 0 3.8 3.8 5.6 8.7 5.6 13.6s-1.9 9.9-5.7 13.7l-94.3 94c-7.6 6.9-19.3 6.7-26.6-.6l-95.7-95.4c-7.5-7.5-7.6-19.7 0-27.3 7.5-7.5 19.7-7.6 27.3 0l81.1 81.9 81-79.9z"
          />
        </svg>
      </div>
    </div>
  </article>
</DocumentFragment>
`;
